flink-doris-connector/src/main/java/org/apache/doris/flink/source/reader/DorisFlightValueReader.java

```diff
@@ -0,0 +1,184 @@
+import org.apache.arrow.adbc.core.AdbcDriver;
+import org.apache.arrow.adbc.core.AdbcException;
+import org.apache.arrow.adbc.core.AdbcStatement;
+import org.apache.arrow.adbc.driver.flightsql.FlightSqlDriver;
+import org.apache.arrow.flight.Location;
+import org.apache.arrow.memory.RootAllocator;
+import org.apache.arrow.vector.ipc.ArrowReader;
+import org.apache.doris.flink.cfg.DorisOptions;
+import org.apache.doris.flink.cfg.DorisReadOptions;
+import org.apache.doris.flink.exception.DorisException;
+import org.apache.doris.flink.exception.IllegalArgumentException;
+import org.apache.doris.flink.exception.ShouldNeverHappenException;
+import org.apache.doris.flink.rest.PartitionDefinition;
+import org.apache.doris.flink.rest.RestService;
+import org.apache.doris.flink.rest.SchemaUtils;
+import org.apache.doris.flink.rest.models.Schema;
+import org.apache.doris.flink.serialization.RowBatch;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+import static org.apache.doris.flink.util.ErrorMessages.SHOULD_NOT_HAPPEN_MESSAGE;
+
+public class DorisFlightValueReader implements ValueReader, AutoCloseable {
+    private static final Logger LOG = LoggerFactory.getLogger(DorisFlightValueReader.class);
+    protected AdbcConnection client;
+    protected Lock clientLock = new ReentrantLock();
+
+    private final PartitionDefinition partition;
+    private final DorisOptions options;
+    private final DorisReadOptions readOptions;
+    private AdbcStatement statement;
+    protected RowBatch rowBatch;
+    protected Schema schema;
+    AdbcStatement.QueryResult queryResult;
+    protected ArrowReader arrowReader;
+    protected AtomicBoolean eos = new AtomicBoolean(false);
+
+    public DorisFlightValueReader(
+            PartitionDefinition partition,
+            DorisOptions options,
+            DorisReadOptions readOptions,
+            Schema schema) {
+        this.partition = partition;
+        this.options = options;
+        this.readOptions = readOptions;
+        this.client = backendClient();
+        this.schema = schema;
+        init();
+    }
+
+    private void init() {
+        clientLock.lock();
+        try {
+            this.statement = this.client.createStatement();
+            this.statement.setSqlQuery(
+                    RestService.parseFlightSql(readOptions, options, partition, LOG));
+            this.queryResult = statement.executeQuery();
+            this.arrowReader = queryResult.getReader();
+        } catch (AdbcException | DorisException e) {
+            throw new RuntimeException(e);
+        } finally {
+            clientLock.unlock();
+        }
+        LOG.debug("Open scan result is, schema: {}.", schema);
+    }
+
+    private AdbcConnection backendClient() {
```

Comment:
Would it be better to change the function name to openConnection?

Reply:
GET